src/python/pants/bsp/protocol.py

```diff
@@ -40,6 +43,12 @@ class _HandlerMapping:
     "workspace/buildTargets": _HandlerMapping(
         WorkspaceBuildTargetsParams.from_json_dict, WorkspaceBuildTargetsResult
     ),
+    "buildTarget/sources": _HandlerMapping(SourcesParams.from_json_dict, SourcesResult),
+    # scala-specific methods
+    # TODO: Discover these via a union so they only load when Scala backend is loaded.
```

Comment:
Mm... so this is the reason for the `Query` then. Ok to punt I suppose. The union would end up defining "additional language specific bsp handlers"?